nomos-services/data-availability/network/src/lib.rs

```diff
@@ -417,11 +432,16 @@ where
     }
 
     fn handle_membership_update(
-        block_numnber: BlockNumber,
-        update: HashMap<<Membership as MembershipHandler>::Id, Multiaddr>,
+        block_number: BlockNumber,
+        update: HashMap<Membership::Id, Multiaddr>,
         storage: &MembershipStorage<StorageAdapter, Membership>,
+        addressbook: &DaAddressbook,
     ) {
-        storage.update(block_numnber, update);
+        storage.update(block_number, update.clone());
+        // since addressbook access is different then membership (get address vs
+        // snapshots) addressbook real implementation would have storage inside
+        // for update and get address
+        addressbook.update(update);
```

Comment:
My idea here is to access and update addressbook in a different them membership. 
The reason is that membership is working on snapshots and the whole snapshots are propagated.

The addressbook update would just need to bulk insert new ids in the underlying storage (rewriting the possible old ones), while `self.addressbook.get_address` would use the underlying storage to get the address.
This is not implemented in the mock implementation, but the real implementation would use a storage to get the address. 
@danielSanchezQ , @bacv Please, let me know what you think.

Reply:
This place for update seems good.